### PR TITLE
Feature: annotate export with texts and arrows

### PR DESCRIPTION
**Pull request by w8r**
Superseeds #52 
connect-arrow-text
adding-arrow-annotation
editing-text-annotation
adding-text-annotation
annotations-bar

### CODE REVIEW

**Review by Leo-Nicolle — APPROVED** [review]

File: src/context/annotations.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+  children: ReactElement;
+}
+
+type AnnotationActionType = "add" | "remove" | "update";
+type AnnotationAction = {
+  type: AnnotationActionType;
+  payload: Annotation;
+};
+
+const annotatationsReducer = (
```
Comment (Leo-Nicolle) [inline src/context/annotations.tsx:61]: tatation !

File: src/utils/index.ts [hunk 2]
```diff
@@ -42,3 +42,58 @@ export function scaleGraph(ogma: Ogma, scale: number) {
...
+  if (!b1) return b0;
+  return [
+    Math.min(b0[0], b1[0]),
+    Math.min(b0[1], b1[1]),
+    Math.max(b0[2], b1[2]),
+    Math.max(b0[3], b1[3]),
+  ];
+}
+
+export function getBoundingBox(ogma: Ogma, texts: boolean): Bounds {
```
Comment (Leo-Nicolle) [inline src/utils/index.ts:60]: I thought I made someting like that in ogma: with https://doc.linkurious.com/ogma/4.4/api.html#NodeList-getBoundingBox and include texts. Is it buggy ?

**Comment by Leo-Nicolle** [comment]
Everything seems to work great ! I could not break it!

File: src/utils/index.ts (hunk 2 above)
Comment (w8r) [reply src/utils/index.ts:60]: yes something stopped me there, maybe it didn't take the texts that were not rendered, into account

Comment (w8r) [reply src/utils/index.ts:60]: Screenshot 2023-05-23 at 12 02 36
I don't know why

**Pull request merged**